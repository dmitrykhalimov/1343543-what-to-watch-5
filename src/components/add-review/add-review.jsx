import React, {useEffect} from "react";
import PropTypes from "prop-types";
import {getActiveFilm} from "../../store/reducers/selectors";
import {withRouter} from "react-router";
import {connect} from "react-redux";
import {validFilm} from "../../utils/props";
import FormReview from "../form-review/form-review";
import Logo from "../logo/logo";
import UserBlock from "../user-block/user-block";
import {fetchSingleFilm} from "../../store/api-actions";

const AddReview = (props) => {
  const id = props.match.params.id;
  const {handlePageLoad, activeFilm} = props;

  const backgroundStyle = {
    backgroundColor: activeFilm.backgroundColor,
  };

  useEffect(() => {
    handlePageLoad(id);
  }, []);

  return (
    <section className="movie-card movie-card--full" style={backgroundStyle}>
      <div className="movie-card__header">
        <div className="movie-card__bg">
          <img src={activeFilm.background} alt={activeFilm.title} />
        </div>

        <h1 className="visually-hidden">WTW</h1>

        <header className="page-header">
          <Logo/>
          <nav className="breadcrumbs">
            <ul className="breadcrumbs__list">
              <li className="breadcrumbs__item">
                <a href="movie-page.html" className="breadcrumbs__link">{activeFilm.title}</a>
              </li>
              <li className="breadcrumbs__item">
                <a className="breadcrumbs__link">Add review</a>
              </li>
            </ul>
          </nav>

          <UserBlock/>
        </header>

        <div className="movie-card__poster movie-card__poster--small">
          <img src={activeFilm.poster} alt={`${activeFilm.title} poster`} width="218" height="327" />
        </div>
      </div>

      <FormReview
        id={id}
      />

    </section>
  );
};

AddReview.propTypes = {
  activeFilm: validFilm,
  handlePageLoad: PropTypes.func.isRequired,
  match: PropTypes.shape({
    params: PropTypes.shape({
      id: PropTypes.string.isRequired,
    }),
  }).isRequired,
};

const mapStateToProps = (state) => ({
  activeFilm: getActiveFilm(state),
});

const mapDispatchToProps = (dispatch) => ({
  handlePageLoad(id) {
    dispatch(fetchSingleFilm(id));
  },
});

export {AddReview};
export default withRouter(connect(mapStateToProps, mapDispatchToProps)(AddReview));

