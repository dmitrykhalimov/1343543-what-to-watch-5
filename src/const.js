export const ratingRanks = [
  {
    'Bad': 2.9
  },
  {
    'Normal': 4.9
  },
  {
    'Good': 7.9
  },
  {
    'Very good': 9.9
  },
  {
    'Awesome': 10
  }
];
