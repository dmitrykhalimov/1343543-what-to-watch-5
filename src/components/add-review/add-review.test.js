import React from "react";
import renderer from "react-test-renderer";
import {AddReview} from "./add-review";
import {TEST_MOCKS, TEST_MOCK_STORE} from "../../const";
import {MemoryRouter} from "react-router-dom";
import configureMockStore from "redux-mock-store";
import {Provider} from "react-redux";

const mockStore = configureMockStore();
const store = mockStore(TEST_MOCK_STORE);

describe(`AddReview`, () => {
  it(`Should AddReview render correctly`, () => {
    const tree = renderer
    .create(
        <Provider store={store}>
          <MemoryRouter>
            <AddReview
              activeFilm={TEST_MOCKS.film}
              match={TEST_MOCKS.match}
              onPageLoad={TEST_MOCKS.noop}
            />,
          </MemoryRouter>
        </Provider>
    )
  .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
