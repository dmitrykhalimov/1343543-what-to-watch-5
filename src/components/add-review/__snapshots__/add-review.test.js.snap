// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddReview Should AddReview render correctly, isFavorite 1`] = `
Array [
  <section
    className="movie-card movie-card--full"
    style={
      Object {
        "backgroundColor": "https://i.imgur.com/1OfiOzs.jpg",
      }
    }
  >
    <div
      className="movie-card__header"
    >
      <div
        className="movie-card__bg"
      >
        <img
          alt="Interstate 60: Episodes of the Road"
          src="#978F7E"
        />
      </div>
      <h1
        className="visually-hidden"
      >
        WTW
      </h1>
      <header
        className="page-header"
      >
        <div
          className="logo"
        >
          <a
            className="logo__link"
            href="/"
            onClick={[Function]}
          >
            <span
              className="logo__letter logo__letter--1"
            >
              W
            </span>
            <span
              className="logo__letter logo__letter--2"
            >
              T
            </span>
            <span
              className="logo__letter logo__letter--3"
            >
              W
            </span>
          </a>
        </div>
        <nav
          className="breadcrumbs"
        >
          <ul
            className="breadcrumbs__list"
          >
            <li
              className="breadcrumbs__item"
            >
              <a
                className="breadcrumbs__link"
                href="movie-page.html"
              >
                Interstate 60: Episodes of the Road
              </a>
            </li>
            <li
              className="breadcrumbs__item"
            >
              <a
                className="breadcrumbs__link"
              >
                Add review
              </a>
            </li>
          </ul>
        </nav>
        <div
          className="user-block"
        >
          <a
            href="/mylist"
            onClick={[Function]}
          >
            <div
              className="user-block__avatar"
            >
              <img
                alt="User avatar"
                height="63"
                src="https://i.kym-cdn.com/photos/images/newsfeed/000/607/429/84b.jpg"
                width="63"
              />
            </div>
          </a>
        </div>
      </header>
      <div
        className="movie-card__poster movie-card__poster--small"
      >
        <img
          alt="Interstate 60: Episodes of the Road poster"
          height="327"
          src="https://m.media-amazon.com/images/M/MV5BZDYwYzlhOTAtNDAwMC00ZTBhLWI4M2QtMTA1NmJhYTdiNTkxXkEyXkFqcGdeQXVyNTM0NTU5Mg@@._V1_UX182_CR0,0,182,268_AL_.jpg"
          width="218"
        />
      </div>
    </div>
    <div
      className="add-review"
    >
      
      <form
        action="#"
        className="add-review__form"
        onSubmit={[Function]}
      >
        <div
          className="rating"
        >
          <div
            className="rating__stars"
          >
            <input
              checked={false}
              className="rating__input"
              id="star-1"
              name="rating"
              onChange={[Function]}
              type="radio"
              value={1}
            />
            <label
              className="rating__label"
              htmlFor="star-1"
            >
              star-1
            </label>
            <input
              checked={false}
              className="rating__input"
              id="star-2"
              name="rating"
              onChange={[Function]}
              type="radio"
              value={2}
            />
            <label
              className="rating__label"
              htmlFor="star-2"
            >
              star-2
            </label>
            <input
              checked={true}
              className="rating__input"
              id="star-3"
              name="rating"
              onChange={[Function]}
              type="radio"
              value={3}
            />
            <label
              className="rating__label"
              htmlFor="star-3"
            >
              star-3
            </label>
            <input
              checked={false}
              className="rating__input"
              id="star-4"
              name="rating"
              onChange={[Function]}
              type="radio"
              value={4}
            />
            <label
              className="rating__label"
              htmlFor="star-4"
            >
              star-4
            </label>
            <input
              checked={false}
              className="rating__input"
              id="star-5"
              name="rating"
              onChange={[Function]}
              type="radio"
              value={5}
            />
            <label
              className="rating__label"
              htmlFor="star-5"
            >
              star-5
            </label>
          </div>
        </div>
        <div
          className="add-review__text"
        >
          <textarea
            className="add-review__textarea"
            id="review-text"
            name="review-text"
            onChange={[Function]}
            placeholder="Review text"
            value=""
          />
          <div
            className="add-review__submit"
          >
            <button
              className="add-review__btn"
              type="submit"
            >
              Post
            </button>
          </div>
        </div>
      </form>
    </div>
  </section>,
  ",",
]
`;
